crates/hir-def/src/body/lower.rs

```diff
@@ -107,7 +112,9 @@ struct ExprCollector<'a> {
     expander: Expander,
     ast_id_map: Arc<AstIdMap>,
     body: Body,
+    krate: CrateId,
     source_map: BodySourceMap,
+    current_try_block: Option<LabelId>,
```

Comment:
This makes me wonder if we should make label resolution be part of the body lowering (not in this PR as it is already way too big, but just a thougght)

Reply:
Fully agree, we should resolve everything we can in body lowering.